mypy/build.py

```diff
@@ -1937,6 +1945,16 @@ def parse_file(self) -> None:
 
         self.check_blockers()
 
+    def parse_inline_configuration(self, source: str) -> None:
+        # Check for inline mypy: options directive and parse them.
+        flags = get_mypy_comments(source)
+        if flags:
+            changes, config_errors = parse_mypy_comments(flags, self.options)
+            self.options = self.options.apply_changes(changes)
+            self.manager.errors.set_file(self.xpath, self.id)
+            for error in config_errors:
+                self.manager.errors.report(-1, 0, error)
```

Comment:
Too bad the line number isn't preserved. If someone accidentally has a `# mypy:` comment in their file, how would they investigate the error if they didn't know about such comments? The error message doesn't even mention `# mypy:` comments.

At the same time, I do understand this is complicated (esp. since the flag parsing is separated from the flag extraction, and shared with `mypy.ini` file parsing).

Perhaps you can parse each `# mypy:` comment separately, and associate the comments with line numbers?

Reply:
Yeah we should do this.